lib/hydra/works/models/concerns/collection_behavior.rb

```diff
@@ -22,6 +22,39 @@ module CollectionBehavior
       type [RDFVocabularies::PCDMTerms.Collection,WorksVocabularies::WorksTerms.Collection]
       include Hydra::Works::AggregatesGenericWorks
       include Hydra::Works::AggregatesCollections
+      include Hydra::Works::BlockChildObjects
+
+      filters_association :members, as: :child_collections, condition: :works_collection?
+      filters_association :members, as: :child_generic_works, condition: :works_generic_work?
+    end
+
+    module ClassMethods
+      def indexer
+        Hydra::Works::CollectionIndexer
+      end
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Collection.
+    def works_collection?
+      true
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic Work.
+    def works_generic_work?
+      false
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic File.
+    def works_generic_file?
+      false
+    end
+
+    def parents
+      aggregated_by
+    end
+
+    def parent_collections
+      aggregated_by.select { |parent| parent.class.included_modules.include?(Hydra::Works::CollectionBehavior) }
```

Comment:
It is not 'new' in the sense that it is refactoring code being ported to Hydra:Works from PCDM.  This entire PR is adding no 'new' code, but is making Hydra:Works consistent with PCDM.  Any issue you raise here applies to PCDM as it exists at master today.